### ISSUE TITLE
'Apache_module' state module crashes instead of gracefully reporting state unavailability

### ISSUE THREAD

**Issue by zer0def** [saltstack/salt#26510]
In Salt 2015.5.2 any attempts to use apache_module states on non-Debian distros results in a report that the state is not found. In 2015.5.3, it crashes in the following manner:
```python-traceback
$ salt-ssh 'centos6' state.highstate
[ERROR   ] JSON Render failed for: 
Traceback (most recent call last):
  File "/tmp/.root_770599__salt/salt-call", line 4, in <module>
    salt_call()
  File "/tmp/.root_770599__salt/salt/scripts.py", line 227, in salt_call
    client.run()
  File "/tmp/.root_770599__salt/salt/cli/call.py", line 69, in run
    caller.run()
  File "/tmp/.root_770599__salt/salt/cli/caller.py", line 236, in run
    ret = self.call()
  File "/tmp/.root_770599__salt/salt/cli/caller.py", line 138, in call
    ret['return'] = func(*args, **kwargs)
  File "/tmp/.root_770599__salt/salt/modules/state.py", line 1154, in pkg
    ret = st_.call_chunks(lowstate)
  File "/tmp/.root_770599__salt/salt/state.py", line 1623, in call_chunks
    running = self.call_chunk(low, running, chunks)
  File "/tmp/.root_770599__salt/salt/state.py", line 1769, in call_chunk
    self._mod_init(low)
  File "/tmp/.root_770599__salt/salt/state.py", line 612, in _mod_init
    self.states['{0}.{1}'.format(low['state'], low['fun'])]  # pylint: disable=W0106
  File "/tmp/.root_770599__salt/salt/utils/lazy.py", line 90, in __getitem__
    raise KeyError(key)
KeyError: 'apache_module.enable'
[ERROR   ] No JSON object could be decoded
```
which could be related to #26454 

The documentation in this case can be considered slightly misleading by not specifying the state module being Debian-specific.

**Comment by jfindlay** [comment]
@zer0def, thanks for the report.  I think there are two issues happening here.  The loader should gracefully fail when it does not find the state, see #25075 for a similar problem.  The apache_module state should work on any system that has apache (more specifically apachectl) installed.  Would you mind filing a separate issue for that?

**Comment by zer0def** [comment]
Sorry for the very late reply.

Well, issue here is that, to my limited knowledge, it's Debian-specific to not necessarily immediately enable modules that are being installed, as opposed to, say, RHEL, which dumps Apache modules to a dir and doesn't do any configuration mangling when a module is installed.

In fact, module management in Debian is done specifically using Perl scripts written by Debian maintainers, not `apachectl`, so there's no need for filling that issue, unless someone's willing to go the extra mile to do something rather invasive (at least in RHEL's approach towards Apache).

**Comment by jfindlay** [comment]
@zer0def, no problem and no need to open a new issue then.  Thanks.

**Comment by rallytime** [comment]
@zer0def This is a regression in the state compiler for 2015.5.3. We've fixed this in 2015.5.6 and 2015.8.0. This bug was fixed with #27681 on 2015.8 branch, which I back-ported to 2015.5 with #27681. Can you give either of those versions/tags a try or  the associated pull request and confirm that this is fixed for you?

**Comment by rallytime** [comment]
Update: It appears that I mis-remembered when my back-porting PR went into the 2015.5 branch and it missed the 2015.5.6 release tag. However, the initial stacktrace reported by @zer0def about the KeyError should be fixed at the HEAD of the 2015.5 branch.

**Comment by rallytime** [comment]
This is fixed at the head of the 2015.5 branch and will be available in the 2015.5.7 release. It's also fixed in the 2015.8.0 releases and newer. Thanks all!
